src/Linter.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace SLLH\ComposerLint;
+
+/**
+ * @author Sullivan Senechal <soullivaneuh@gmail.com>
+ */
+final class Linter
+{
+    /**
+     * @param array $manifest composer.json file manifest
+     *
+     * @return string[]
+     */
+    public function validate($manifest)
+    {
+        $errors = array();
+
+        if (isset($manifest['config']['sort-packages']) && $manifest['config']['sort-packages']) {
+            foreach (array('require', 'require-dev', 'conflict', 'replace', 'provide', 'suggest') as $linksSection) {
+                if (array_key_exists($linksSection, $manifest) && !$this->packagesAreSorted($manifest[$linksSection])) {
+                    array_push($errors, 'Links under '.$linksSection.' section are not sorted.');
+                }
+            }
+        }
+
+        return $errors;
+    }
+
+    private function packagesAreSorted(array $packages)
+    {
+        $names = array_keys($packages);
+
+        $hasPHP = in_array('php', $names);
+        $extNames = array_filter($names, function ($name) {
+            return 'ext-' === substr($name, 0, 4) && !strstr($name, '/');
```

Comment:
> this makes the code unclear (at least to coders that don't come from Dagobah)

Habits! :trollface: 

> we've seen that the main reason they have for closing an RFC for a change in coding style [...]  if they make a wrong decision and know it, they are going to stick with it forever

For this case, they never tell this is a bad decision. This is bad for your point of view. For them, this is a decision.

The "merge nightmare" is an additional reason.

> I don't think it makes sense to follow SF coding standard (some of them, but not all of them), especially in this package that is not an SF bundle.

I decide to follow SF standard on each of my project to not balance between them.

Agree or not, SF coding standard includes Yoda condition, so we will follow it, even for sonata or my personal projects because I choose SF standards. :wink: